hazelcast/src/main/java/com/hazelcast/spi/discovery/integration/DiscoveryService.java

```diff
@@ -83,4 +86,31 @@ public interface DiscoveryService {
      * @since 3.7
      */
     Map<String, String> discoverLocalMetadata();
+
+    /**
+     * Marks the passed {@link Address} as unhealthy, which prevents it from being offered as a
+     * viable endpoint in some {@link DiscoveryStrategy} implementations, usually prompting
+     * this endpoint to be periodically probed for liveliness. If not supported by the underlying
+     * implementation, then this call does nothing.
+     *
+     * @see DiscoveryStrategy#markEndpointAsUnhealthy(Address)
+     *
+     * @param address the address to mark as unhealthy
+     * @since 5.4
+     */
+    default void markEndpointAsUnhealthy(Address address) { }
```

Comment:
@hazelcast/cloud-native do you have any objections to these 2 additional methods in the `DiscoveryService` SPI?

Reply:
It seems fine, especially since it can be a no-op